### ISSUE TITLE
Missed translation units for language Spanish on file EmailsBody.es-ES.xlf

### ISSUE THREAD

**Issue by jcarloscid** [PrestaShop/PrestaShop#16849]
**Describe the bug**
Some translation units are missed on the file EmailsBody.es-ES.xlf as compared to default EmailsBody.xlf

So far I have found the following ones by reviewing the emails on Spanish:
file "mails/themes/modern/core/outofstock.html.twig"
trans-unit id="e57e950cfde3e24c3674f9266753edec"
trans-unit id="dd099f9c8d04d30fa277471417a372be"

file "mails/themes/modern/core/password_query.html.twig" 
trans-unit id="3516ed9d771d15439e2ac9d3bf81ad84"
trans-unit id="87a8bbd45aa2890a2929c07d49fa64f1"
trans-unit id="0b0df2fff267640acf7f0d03a9b90aa6"

This is causing the incomplete translation of some emails (the first one affectes several templates, the second one is just for the  password_query email.

**To Reproduce**
This happened after migrating a shop in PS 1.7.5.1 to PS 1.7.6.2 using the 1-click upgrade module.

**Additional information**
PrestaShop version: 1.7.6.2

I am attaching the file installed after the upgrade.
[EmailsBody.es-ES.xlf.zip](https://github.com/PrestaShop/PrestaShop/files/3978499/EmailsBody.es-ES.xlf.zip)

**Comment by khouloudbelguith** [comment]
Hi @jcarloscid,

Have you the same issue with this ticket: https://github.com/PrestaShop/PrestaShop/issues/16555?

Thanks!

**Comment by jcarloscid** [comment]
I haven´t tried that.

I discover the problem after exporting the emails in Spanish replacing existing templates.

Looking at the emails generated I found this translations units where used by emails but missed in Spanish file, so I added them manually and repeat the email generation.

**Comment by khouloudbelguith** [comment]
@jcarloscid, thanks for your feedback.
>I discover the problem after exporting the emails in Spanish

How did you export the emails in Spanish? from the BO?

Thanks!

**Comment by jcarloscid** [comment]
Yes. using the new option on Design --> Email Themes --> Generate Emails 

![image](https://user-images.githubusercontent.com/35863705/71160145-4ee6ea80-2247-11ea-8cf9-5214a94d5947.png)

(I'm sorry the user interface is in Spanish)

I plan to upgrade another shop (same initial and final versions) in the following days. If it helps, I can capture some of the emails with mixed translated/non-translated content.

However, if you try to find the reported translation units on /app/Resources/translations/default/EmailsBody.xlf and the same units on /app/Resources/translations/es-ES/EmailsBody.es-ES.xlf, are you able to find them in both?

Because I think that this is not a problem with a particular operation or version, but something lost on the source code. I am basing this assumption on the current content of these two files:

https://github.com/PrestaShop/PrestaShop/blob/develop/app/Resources/translations/default/EmailsBody.xlf
https://github.com/PrestaShop/TranslationFiles/blob/master/1.7/translations/1.7.6/es-ES/EmailsBody.es-ES.xlf

None of the keys:
"e57e950cfde3e24c3674f9266753edec"
"dd099f9c8d04d30fa277471417a372be"
"3516ed9d771d15439e2ac9d3bf81ad84"
"87a8bbd45aa2890a2929c07d49fa64f1"
"0b0df2fff267640acf7f0d03a9b90aa6"
are present on the second, but all of them are present on the first one.

Sorry, may be this issue is really for the TranslationFiles repository. I am not familiar on how you organize the resolution of this kind of issues.

**Comment by khouloudbelguith** [comment]
@jcarloscid, we cannot edit the translation which is in the core of the mails.
It is planned for PS1.7.7.7: https://github.com/PrestaShop/PrestaShop/pull/14809

Thanks to check & feedback.

**Comment by jcarloscid** [comment]
Don´t know if your answer really address the issue, but I'm happy to hear that.

By the time that you implement the editor, you might want to fix the sources.

**Comment by khouloudbelguith** [comment]
@jcarloscid, sorry, but what exact sources?

Thanks!

**Comment by jcarloscid** [comment]
My workaround for the issue it was as follows:
1. Edit EmailsBody.es-ES.xlf and add the missed translation units using copy&paste from EmailsBody.xlf
2. Translate new units to Spanish
3. Clear cache (!important)
3. Regenerate emails in Spanish overriding existing ones 

I am attaching the modified file which is located at /app/Resources/translations/es-ES
[EmailsBody.es-ES.xlf.zip](https://github.com/PrestaShop/PrestaShop/files/3983236/EmailsBody.es-ES.xlf.zip)

**Comment by jcarloscid** [comment]
@khouloudbelguith /app/Resources/translations/es-ES/EmailsBody.es-ES.xlf

**Comment by khouloudbelguith** [comment]
@jcarloscid, here's my two files with PS1.7.6.2.

[EmailsBody.xlf.txt](https://github.com/PrestaShop/PrestaShop/files/3984561/EmailsBody.xlf.txt)
[EmailsBody.es-MX.xlf.txt](https://github.com/PrestaShop/PrestaShop/files/3984571/EmailsBody.es-MX.xlf.txt)

Those keys are present in EmailsBody.es-MX.xlf file:
```

"e57e950cfde3e24c3674f9266753edec"
"dd099f9c8d04d30fa277471417a372be"
"3516ed9d771d15439e2ac9d3bf81ad84"
"87a8bbd45aa2890a2929c07d49fa64f1"
"0b0df2fff267640acf7f0d03a9b90aa6"
```

Thanks!

**Comment by jcarloscid** [comment]
Please, @khouloudbelguith check my original message. The locale I sent was es-ES. I haven't check this issue in any other locale.

And please also take a look at this one, 

https://github.com/PrestaShop/TranslationFiles/blob/master/1.7/translations/1.7.6/es-ES/EmailsBody.es-ES.xlf

Kind regards.

**Comment by khouloudbelguith** [comment]
@jcarloscid, Thanks for these clarifications
I have the same issue with PS1.7.6.2.
[EmailsBody.es-ES.xlf.TXT](https://github.com/PrestaShop/PrestaShop/files/3984754/EmailsBody.es-ES.xlf.TXT)
PIng @jolelievre what do you think?

Thanks!

**Comment by hibatallahAouadni** [comment]
Ping @PrestaShop/prestashop-core-developers friendly reminder :wink:

**Comment by hibatallahAouadni** [comment]
Hello @jcarloscid 

Do you still encounter the same issue with PS 8.1.x?
Please check and feedback.

Thanks!

**Comment by djoelleuch** [comment]
Hello @jcarloscid ,
I think that this issue is fixed with https://github.com/PrestaShop/PrestaShop/pull/24049.
and Since we had no news from you for more than 20 days, I'll close this ticket to avoid cluttering up the backlog. Please note that you can always create a new one if further information pops up.

Thank you

**Comment by jcarloscid** [comment]
I am sorry I am not using that PS versión yet, so I cannot test myself.KREl 22 sept 2023 15:28, Hejer ELLEUCH ***@***.***> escribió:
Hello @jcarloscid ,
I think that this issue is fixed with #24049.
and Since we had no news from you for more than 20 days, I'll close this ticket to avoid cluttering up the backlog. Please note that you can always create a new one if further information pops up.
Thank you
